src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java

```diff
@@ -258,18 +262,19 @@ private void addStrandBiasAnnotations(final ReadLikelihoods<Allele> likelihoods,
      * @return                                      genotype likelihoods for homRef and het for each alternate allele
      */
     private PerAlleleCollection<Double> getHetGenotypeLogLikelihoods(final ReadLikelihoods<Allele> likelihoods,
-                                                                     final String sampleName,
+                                                                     final String sampleNameForAlleleFractions,
+                                                                     final String sampleNameForLikelihoods,
                                                                      final OptionalDouble givenAltAlleleFraction,
                                                                      final Strand strand) {
-        final PerAlleleCollection<Double> alleleFractions = getAlleleFractions(likelihoods, sampleName, strand);
+        final PerAlleleCollection<Double> alleleFractions = getAlleleFractions(likelihoods, sampleNameForAlleleFractions, strand);
```

Comment:
We should not compute `alleleFractions` if `givenAltAlleleFraction ` is not empty.